A15-binary-logging.md

```diff
@@ -0,0 +1,375 @@
+# gRPC Binary Logging
+
+*   Author(s): Yuchen Zeng, Spencer Fang
+*   Approver: a11r
+*   Status: In Review
+*   Implemented in:
+*   Last updated: 07/18/2018
+*   Discussion at:
+
+## Abstract
+
+Provide binary logging functionality in gRPC.
+
+## Background
+
+gRPC binary logging logs RPCs in binary format. RPC logs generated from
+production traffic will be useful in the following cases:
+
+-   Troubleshooting services, finding exceptions
+-   Loadtesting
+-   Replaying RPCs from production
+
+This functionality should be added in each of our supported languages. The ease
+of adoption should also be considered.
+
+## Proposal
+
+The proposed design is adding binary logging as a built-in functionality in
+gRPC. Users can turn it on/off by setting the `GRPC_BINARY_LOG_FILTER`
+environment variable or a similarly named flag or option string.
+Turning on binary logging should be a configuration setting and require no
+code changes. The control interface should allow users to change the amount
+and contents of logs.
+
+The logged events correspond to what is visible at the application
+level, and as a result they exclude things like [hedging and
+retries](./A6-client-retries.md).
```

Comment:
Maybe we should define what it means to "exclude" retries.  Does this mean that we should only log events from the committed attempt?

Reply:
Yes. I will add this statement to the doc.